src/DataModel/IData.ts

```diff
@@ -0,0 +1,6 @@
+import {IMetadata} from "./IMetadata";
+
+export interface IData extends Object
```

Comment:
I don't like this and `IMetadata`. They seems too generic and separated from the Hydra terminology. I find using such disconnected terms confusing.

For `IData`, did you mean something like `IHydraApi`?